### PR TITLE
Label namespaces with lagoon.sh/environmentType

### LINKED ISSUE #1364: Refactor Container Logs Elasticsearch Index Name creation to generate less indexes
Currently we just use the Kubernetes Namespace name as the indicator for index names of container logs inside elasticsearch: `container-logs-${record['namespace_name']}-${Time.at(time).strftime("%Y.%m")}`

The problem is that with having many PR environments we create a new index for each PR environment even though they only have a couple of hundred log entries. This makes the elasticsearch slow.

Therefore we would like to change the index name to separate between the environment type (where we currently have `production` and `development`):
- `container-logs-${lagoon-projectname}-${lagoon-environmenttype}-${Time.at(time).strftime("%Y.%m")}`

In order to do this we're missing two essential pieces in the payload of the container logs:
- lagoon project name
- lagoon environment type

Fluentd has the possibility to inject the labels of the namespace into the payload of the container logs, therefore the plan is:

- [ ] Inject the Lagoon ProjectName, EnviornmentName, EnvironmentType as a label into the Kubernetes Namespace when we create the namespace, but also when we redeploy (in order to handle namespaces that don't have the labels yet, but also in case the environment type change)
- [ ] create a script that add/updates the labels in all the namespaces created by lagoon
- [ ] Update the Fluentd config https://github.com/amazeeio/lagoon/blob/master/services/logs-forwarder/.lagoon.multi.yml#L223-L228 to check for these new variables and if they exist use them instead of the current way (but provide a fallback to use the old system, plus also for namespaces that are not created by lagoon - like `default`)

### PR DESCRIPTION
**Pull request by smlx**
# Checklist
- [x] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [x] PR title is ready for changelog and subsystem label(s) applied

This PR makes information about Lagoon environment types available in the Kubernetes API. This facilitates, among other things, a fix for #1364.

# Closing issues
In combination with #1959, closes #1364.

### CODE REVIEW

**Comment by smlx** [comment]
Converted to draft as I'm not sure the openshift labels are being applied

File: services/kubernetesbuilddeploy/src/index.ts [hunk 1]
```diff
@@ -318,7 +318,8 @@ const messageConsumer = async msg => {
           "name":openshiftProject,
           "labels": {
             "lagoon.sh/project": projectName,
-            "lagoon.sh/environment": environmentName
+            "lagoon.sh/environment": environmentName,
+            "lagoon.sh/environmenttype": lagoonEnvironmentType
```
Comment (Schnitzel) [inline services/kubernetesbuilddeploy/src/index.ts]:
```suggestion
            "lagoon.sh/environmentType": lagoonEnvironmentType
```

File: services/openshiftbuilddeploy/src/index.ts [hunk 2]
```diff
@@ -333,7 +333,8 @@ const messageConsumer = async msg => {
             "name":openshiftProject,
             "labels": {
               "lagoon.sh/project": safeProjectName,
-              "lagoon.sh/environment": safeBranchName
+              "lagoon.sh/environment": safeBranchName,
+              "lagoon.sh/environmenttype": lagoonEnvironmentType
```
Comment (Schnitzel) [inline services/openshiftbuilddeploy/src/index.ts]:
```suggestion
              "lagoon.sh/environmentType": lagoonEnvironmentType
```
we use already this way in https://github.com/amazeeio/lagoon/blob/6d08071fad9619d9fe2bd54a77f18124a66c9691/images/kubectl-build-deploy-dind/helmcharts/cli/templates/_helpers.tpl#L58

**Review by Schnitzel — CHANGES_REQUESTED** [review]

File: services/kubernetesbuilddeploy/src/index.ts (hunk 1 above)
Comment (smlx) [reply services/kubernetesbuilddeploy/src/index.ts]: :+1:

File: services/openshiftbuilddeploy/src/index.ts (hunk 2 above)
Comment (smlx) [reply services/openshiftbuilddeploy/src/index.ts]: :+1:

**Comment by smlx** [comment]
Setting this as ready for review. Openshift labelling doesn't work, but that's covered in https://github.com/amazeeio/lagoon/issues/1969.

**Pull request merged**